libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenuinputs/index.tsx

```diff
@@ -118,15 +220,53 @@ export const FloatingActionMenuInputs = (props: FloatingActionMenuInputsProps):
     return <TextField className="msla-dynamic-added-param-value-TextField" value={description} onChange={onDescriptionChange} />;
   };
 
+  const stringListValues = (schemaKey: string): string[] => {
+    const model = dynamicParameterModels.find((model) => model.schemaKey === schemaKey);
+    if (model && (model.properties as DynamicallyAddedParameterInputsTextProperties).enum) {
+      return (model.properties as DynamicallyAddedParameterInputsTextProperties).enum;
+    }
+    if (model && (model.properties as DynamicallyAddedParameterInputsArrayProperties).items) {
+      return (model.properties as DynamicallyAddedParameterInputsArrayProperties).items.enum;
+    }
+    return [];
+  };
+
+  const getStringListUpdateHandler = (schemaKey: string, newValue: string[]) => {
+    if (isDynamicParameterDropdown(schemaKey)) {
+      return onStringDropdownListUpdate(schemaKey, newValue);
+    }
+    if (isDynamicParameterMultiSelect(schemaKey)) {
+      return onStringMultiSelectListUpdate(schemaKey, newValue);
+    }
+    return undefined;
+  };
+
   const dynamicParameterModels: DynamicallyAddedParameterInputsModel[] = deserialize(
     props.initialValue,
     props.isRequestApiConnectionTrigger
   ).map((model) => ({
     ...model,
     onTitleChange: onDynamicallyAddedParameterTitleChange,
     onRequiredToggle: onDynamicallyAddedParameterRequiredToggle,
+    onStringDropdownListToggle: model.properties.type === 'string' ? onStringDropdownListToggle : undefined,
+    onStringMultiSelectListToggle: ['array', 'string'].includes(model.properties.type) ? onStringMultiSelectListToggle : undefined,
     onDelete: onDynamicallyAddedParameterDelete,
     onRenderValueField,
+    isDynamicParameterMultiSelect,
+    isDynamicParameterDropdown,
+    onStringListUpdate: getStringListUpdateHandler,
+    shouldDisplayAddDropdownOption:
+      (model.properties['x-ms-content-hint'] === 'TEXT' && !(model.properties as DynamicallyAddedParameterInputsTextProperties).enum) ??
+      undefined,
+    shouldDisplayAddMultiSelectOption:
+      (model.properties['x-ms-content-hint'] === 'TEXT' && !(model.properties as DynamicallyAddedParameterInputsArrayProperties).items) ??
+      undefined,
+    shouldDisplayRemoveDropdownOption:
+      (model.properties['x-ms-content-hint'] === 'TEXT' && !!(model.properties as DynamicallyAddedParameterInputsTextProperties).enum) ??
+      undefined,
+    shouldDisplayRemoveMultiSelectOption:
+      (model.properties.type === 'array' && !!(model.properties as DynamicallyAddedParameterInputsArrayProperties).items) ?? undefined,
+    stringListValues: ['array', 'string'].includes(model.properties.type) ? stringListValues : undefined,
```

Comment:
Looks like `['array', 'string'].includes(model.properties.type)` is also reused a number of times?